src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/blob.py

```diff
@@ -266,9 +268,18 @@ def _copy_blob_to_blob_container(blob_service, source_blob_service, destination_
 
 def _copy_file_to_blob_container(blob_service, source_file_service, destination_container,
                                  source_share, source_sas, source_file_dir, source_file_name):
-    source_file_dir = source_file_dir if source_file_dir else None
-    file_url = source_file_service.make_file_url(source_share, source_file_dir, source_file_name,
-                                                 sas_token=source_sas)
+    from six.moves.urllib.parse import quote, urlparse, urlunparse  # pylint: disable=import-error
+    try:
+        file_url = source_file_service.make_file_url(source_share, source_file_dir or None, source_file_name,
+                                                     sas_token=source_sas)
+    except UnicodeEncodeError:
+        source_file_dir = source_file_dir.encode('utf-8')
+        source_file_name = source_file_name.encode('utf-8')
+        file_url = source_file_service.make_file_url(source_share, source_file_dir or None, source_file_name,
+                                                     sas_token=source_sas)
+    url_parts = urlparse(file_url)
+    quoted_path = quote(url_parts.path, '/()$=\',~')
+    file_url = urlunparse(url_parts[:2] + (quoted_path,) + url_parts[3:])
```

Comment:
Can we put these 3 lines in a storage util function or something?
It looks like the same thing over and over.
e.g. there is sdkutil.py or storage_url_helpers.py?

Reply:
And also make `'/()$=\',~'` a constant in this util file.